File: Components/WinForms/DevExpress.HybridApp.Win/CS/DevExpress.HybridApp.Win/ViewModels/Employee/EmployeeMailMergeViewModel.cs
using DevExpress.DevAV.Common.DataModel;
using DevExpress.Mvvm.POCO;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace DevExpress.DevAV.ViewModels {
    public class EmployeeMailMergeViewModel : MailMergeViewModel<Employee, object> {
        protected EmployeeMailMergeViewModel(IEnumerable<Employee> entities = null, Employee selectedEntity = null, string selectedTemplateName = null, object linksViewModel = null)
            : base(entities, selectedEntity, selectedTemplateName, linksViewModel) {
        }

        public static EmployeeMailMergeViewModel Create<TUnitOfWork, TPrimaryKey>(IUnitOfWorkFactory<TUnitOfWork> unitOfWorkFactory, Func<TUnitOfWork, IRepository<Employee, TPrimaryKey>> getRepositoryFunc, TPrimaryKey? key, string selectedTemplateName = null, object linksViewModel = null)
            where TUnitOfWork: IUnitOfWork
            where TPrimaryKey: struct {
            var repository = getRepositoryFunc(unitOfWorkFactory.CreateUnitOfWork());
            var entities = repository.ToArray();
            var selectedEntity = key != null ? repository.Find(key.Value) : null;
            return ViewModelSource.Create(() => new EmployeeMailMergeViewModel(entities, selectedEntity, selectedTemplateName, linksViewModel));
        }
    }
}

File: Components/WinForms/DevExpress.OutlookInspiredApp.Win/VB/DevExpress.OutlookInspiredApp.Win/ViewModels/Employee/EmployeeMailMergeViewModel.vb
Imports DevExpress.Mvvm.POCO
Imports DevExpress.DevAV.ViewModels

Namespace DevExpress.DevAV.ViewModels

    Public Class EmployeeMailMergeViewModel
        Inherits MailMergeViewModelBase(Of EmployeeMailTemplate)

        Public Shared Function Create() As EmployeeMailMergeViewModel
            Return ViewModelSource.Create(Function() New EmployeeMailMergeViewModel())
        End Function
        Protected Sub New()
        End Sub
    End Class
End Namespace
